src/init.cpp

```diff
@@ -1805,6 +1805,9 @@ bool AppInitMain(boost::thread_group& threadGroup, CScheduler& scheduler)
                 pcoinsTip.reset();
                 pcoinsdbview.reset();
                 pcoinscatcher.reset();
+                // new CBlockTreeDB tries to delete the existing file, which
+                // fails if it's still open from the previous loop. Close it first:
+                pblocktree.reset();
```

Comment:
Should apply the same fix to `evoDb` and `deterministicMNManager` below I think.

Reply:
https://github.com/dashpay/dash/pull/3361/commits/6c61159a9911ab5e9ed2fe1528b4d92557439d95